circle.yml

```diff
@@ -1,28 +1,63 @@
 machine:
   environment:
     # We need to set this variable to let python see the package installed through apt-get
-    PYTHONPATH: "/usr/lib/python2.7/dist-packages"
-dependencies:
-  cache_directories:
+    # PYTHONPATH: "/usr//usr/lib/python2.7/dist-packages"
+    PYTHON_VERSION: "2.6"
+    NUMPY_VERSION: "1.6.2"
+    SCIPY_VERSION: "0.12.0"
+    CYTHON_VERSION: "0.21"
+dependencies: 
+  cache_directories: 
     - "~/scikit_learn_data"
     - "~/scikit-learn.github.io"
+    - "~/download"
   # Various dependencies
-  pre:
+  pre: 
+    - |
+        # deactivate circleci virtualenv and setup miniconda env.
+        deactivate
+        # Use the miniconda installer for faster download / install of conda
+        # itself
+        pushd .
+        cd
+        mkdir -p download
+        cd download
+        echo "Cached in $HOME/download :"
+        ls -l
+        echo
+        if [[ ! -f miniconda.sh ]]
+           then
+           wget http://repo.continuum.io/miniconda/Miniconda-3.6.0-Linux-x86_64.sh \
+           -O miniconda.sh
+        fi
+        chmod +x miniconda.sh && ./miniconda.sh -b -p $HOME/miniconda
+        cd ..
+        export PATH="$HOME/miniconda/bin:$PATH"
+        conda update --yes conda
+        popd
+        
+        # Configure the conda environment and put it in the path using the
+        # provided versions
+        conda create -n testenv --yes python=$PYTHON_VERSION \
+        numpy=$NUMPY_VERSION scipy=$SCIPY_VERSION cython=$CYTHON_VERSION \
+        nose coverage matplotlib sphinx
+        source /home/ubuntu/miniconda/envs/testenv/bin/activate testenv
+    - echo $PATH
+    - |
+        binList="$(ls /home/ubuntu/miniconda/envs/testenv/bin)"
+        echo "${binList}"
     - sudo apt-get update
     - sudo apt-get install libatlas-dev libatlas3gf-base
     - sudo apt-get install build-essential python-dev python-setuptools
-    - sudo apt-get install python-numpy python-scipy python-dev python-matplotlib
-    - sudo apt-get install python-nose python-coverage
-    - sudo apt-get install python-sphinx
     - pip install cython
   # The --user is needed to let sphinx see the source and the binaries
   # The pipefail is requested to propagate exit code
-  override:
+  override: 
     - python setup.py install --user
```

Comment:
Also instead of using `python install --user` which will probably won't work correctly inside a `conda env` we should probably use `python setup.py develop` to install scikit-learn in development mode.